### DISCUSSION TITLE
Surface `Drop` order

### DISCUSSION THREAD

**Discussion by sotrh** [gfx-rs/wgpu#3991] [Q&A] [answered]
Had [a question](https://github.com/sotrh/learn-wgpu/issues/475) on my tutorial regarding whether the `wgpu::Surface` calls any native APIs that require the `winit::Window` to still be alive when it gets dropped. I don't know the answer to that question, so I decided to ask here.

**Comment by JustAnotherCodemonkey** [comment, marked as answer]
This could be wrong as I haven't actually looked at the code but to my understanding, the Surface contains several different unsafe references to the window that assume that it is still alive. This includes a raw pointer to the underlying memory in VRAM where the window's pixel buffer resides. Trying to, say, render to this underlying texture after the window has closed and the memory may be freed would be Undefined Behavior. There's a reason why creating a WGPU Surface from a Winit Window is considered Unsafe. 

In short, if a Surface persists after its window becomes invalid, any operations involving it will be Undefined Behavior.

**Reply by JustAnotherCodemonkey** [reply]
Oh and btw I love your tutorial. I must say though, it is a bit lacking in terms of stuff on compute shaders. Maybe I can help out with that as I've done quite a bit of research and screwing around on that subject.

**Reply by sotrh** [reply]
Thanks for your answer. As far as helping on the tutorial itself, I haven't decided on how I would handle guest articles, so I'll pass for now.

**Reply by JustAnotherCodemonkey** [reply]
No problem and that's perfectly understandable on that second part. I'm writing my own mdbook about wgpu as well
